lib/src/main/java/com/auth0/android/lock/Lock.java

```diff
@@ -150,8 +151,8 @@ private void processEvent(@NonNull Context context, @NonNull Intent data) {
         switch (action) {
             case Constants.AUTHENTICATION_ACTION:
                 Log.v(TAG, "AUTHENTICATION action received in our BroadcastReceiver");
-                if (data.hasExtra(Constants.ERROR_EXTRA)) {
-                    callback.onError(new LockException(data.getStringExtra(Constants.ERROR_EXTRA)));
+                if (data.hasExtra(Constants.EXCEPTION_EXTRA)) {
+                    callback.onError(new LockException((AuthenticationException) data.getSerializableExtra(Constants.EXCEPTION_EXTRA)));
```

Comment:
Is the deserialization into an `AuthenticationException` expected to always succeed?

Reply:
nvm, I saw below that the AuthenticationException is sent by the library.